theano/compile/mode.py

```diff
@@ -290,7 +290,10 @@ def __str__(self):
 
     def __get_optimizer(self):
         if isinstance(self._optimizer, gof.Query):
-            return optdb.query(self._optimizer)
+            if hasattr(self, '_optimizer_cache'):
+                return self._optimizer_cache
+            self._optimizer_cache = optdb.query(self._optimizer)
+            return self._optimizer_cache
```

Comment:
Are there any config flags that could change during the execution, and that could have an impact on that?
If so, we could just prevent it from being modified during the execution.